skimage/segmentation/_quickshift.pyx

```diff
@@ -82,23 +48,21 @@ def quickshift(image, ratio=1.0, kernel_size=5, max_dist=10,
     # an effect for very high max_dist.
 
     # window size for neighboring pixels to consider
-    if kernel_size < 1:
-        raise ValueError("Sigma should be >= 1")
     cdef float kernel_size_sq = kernel_size**2
-    cdef int w = np.ceil(3 * kernel_size)
+    cdef int w = cnp.ceil(3 * kernel_size)
 
-    cdef Py_ssize_t height = image_c.shape[0]
-    cdef Py_ssize_t width = image_c.shape[1]
-    cdef Py_ssize_t channels = image_c.shape[2]
+    cdef Py_ssize_t height = image.shape[0]
+    cdef Py_ssize_t width = image.shape[1]
+    cdef Py_ssize_t channels = image.shape[2]
     cdef double current_density, closest, dist
 
     cdef Py_ssize_t r, c, r_, c_, channel, r_min, c_min
 
-    cdef cnp.float_t* image_p = <cnp.float_t*> image_c.data
+    cdef cnp.float_t* image_p = <cnp.float_t*> image.data
     cdef cnp.float_t* current_pixel_p = image_p
 
     cdef cnp.ndarray[dtype=cnp.float_t, ndim=2] densities \
```

Comment:
This could be a memoryview as well I assume (on my phone so cannot check all the code).